api/envoy/api/v2/cds.proto

```diff
@@ -234,6 +234,26 @@ message Cluster {
   //   verification.
   auth.UpstreamTlsContext tls_context = 11;
 
+  // TransportSocketMatch specifies what transport socket config will be used
+  // when the matches conditions are satisfied.
+  message TransportSocketConfig {
+    // The name of the match.
+    string name = 1;
+    // The configuration of the transport socket.
+    core.TransportSocket transport_socket = 3;
+  }
+
+  // Configuration to use different transport sockets for different endpoints.
+  // This should only be used for EDS typed cluster.
```

Comment:
why only eds? we add the tls object name to the LbEndpoint. Not just eds. There are so many use cases for static clusters